Automattic/wp-calypso
# Staging sites: Remove staging banner

Fixes https://github.com/Automattic/dotcom-forge/issues/7948

## Proposed Changes

* Remove the yellow and black banner that's displayed on staging sites.
* Remove the icon next to the `Staging Site` header.
* Clean up the header styles so they match when switching between production and staging sites.

| Before | After |
|--------|--------|
| <img width="1728" alt="Screenshot 2024-07-11 at 16 48 31" src="https://github.com/Automattic/wp-calypso/assets/417538/46e5454c-4bc5-4cc2-bc1a-73175c7862b5"> | <img width="1728" alt="Screenshot 2024-07-11 at 16 48 03" src="https://github.com/Automattic/wp-calypso/assets/417538/5a83f25c-7dac-4f4e-850f-114890b89a5a"> |

## Why are these changes being made?


* We now display a yellow border and buttons when viewing a staging site, so we don't need to signal that status from within the `Staging Site` tab.

## Testing Instructions



* Run with `yarn start`
* Create a site
* Navigate to the `Staging Site` tab
* Click the `Add staging site` button
* Switch to the staging site and note that there yellow and black banner is no longer shown on the `Staging Site` tab.

## Pre-merge Checklist
